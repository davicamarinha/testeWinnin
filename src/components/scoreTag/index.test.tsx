import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import ScoreTag from "./";

describe("ScoreTag Tests", () => {
  it("renders a snapshot", () => {
    const { container } = render(<ScoreTag value={50} />);
    expect(container).toMatchSnapshot();
  });

  it("should render red below 50", () => {
    const { queryByTestId } = render(<ScoreTag value={49} />);
    expect(queryByTestId("score")).toHaveClass("bg-asLow");
  });

  it("should render yellow when the value is over or equal 50 and below 80", () => {
    const { queryByTestId } = render(<ScoreTag value={60} />);
    expect(queryByTestId("score")).toHaveClass("bg-asMedium");
  });

  it("should render green when the value is over 80", () => {
    const { queryByTestId } = render(<ScoreTag value={90} />);
    expect(queryByTestId("score")).toHaveClass("bg-asGood");
  });

  it("should render 0% if value is null", () => {
    render(<ScoreTag value={null} />);
    expect(screen.getByText("0%")).toBeInTheDocument();
  });
});
