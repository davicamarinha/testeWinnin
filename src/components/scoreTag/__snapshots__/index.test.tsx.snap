// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreTag Tests renders a snapshot 1`] = `
<div>
  <div
    class="px-[8px] py-[3px] rounded-[4px] text-white text-2xl font-normal bg-asMedium"
    data-testid="score"
  >
    50
    %
  </div>
</div>
`;
